### DISCUSSION TITLE
What about making everything public instead of protected and private ?

### DISCUSSION THREAD

**Discussion by murataka** [Sixze/ALS-Refactored#487] [General]
There are some functionalities made private in source , and whenever we want to make changes to plugin , how it behaves , the options are limited to 

1. Edit configuration data assets ,or the exposed settings in Als classes
2. Alter the sourcecode , which brings hard work when we want to upgrade the plugin to new version.

What I currently encounter is , when I use live retargeting , some animations , like mantling does not fit when the skeleton is little bigger , for example. To overcome that, I was digging into the source , to find where I can override mantling calculation functions , or pass the overlay skeleton instead of default mesh skeleton to related functions. However , I see that some functions are private and so is not overridable. I hesitated to make changes to source , assuming there's a better way to handle that without altering sourcecode. Maybe retargeting animations onto a different skeleton , create a custom control rig etc, is the solution ?, but that would still require changes to original plugin code, say for each character .

Am I missing something ? 

To support different skeletons is possible without altering original sourcecode ?
 If yes, how ?
 If no , what do you think about making functions public ( as virtual to make override work) where possible, and use getter -setter methods for properties ,which are also overridable ?

**Comment by Sixze** [comment]
ALS is more of a template than a plugin, and is expected to be modified to suit the requirements of a particular project.

**Comment by ameaninglessname** [comment]
I think we will maintain our own fork, and rebase it to latest updates if needed.

**Reply by Magic-Summer** [reply]
Do you have any best practice for this? E.g. how often (or when) do you rebase onto the @Sixze version?

**Reply by ameaninglessname** [reply]
Ideally,I will always sync to the latest commit. In reality, I do that about once a week.
